src/tuner/Makefile.am

```diff
@@ -15,7 +15,8 @@ if !ENABLE_NEURON
 
 sources += \
 	nccl_ofi_regions.c \
-	nccl_ofi_tuner.c
+	nccl_ofi_tuner.c \
+	../nccl_ofi_system.c
```

Comment:
try: `$(abs_top_srcdir)/src/nccl_ofi_system.c`

Reply:
seems like that didn't take -- 

https://www.gnu.org/software/automake/manual/html_node/Program-and-Library-Variables.html#index-maude_005fSOURCES

We specify subdir-objects globally in the automake init.

> By default the output file (on Unix systems, the .o file) will be put into the current build directory. However, if the option subdir-objects is in effect in the current directory then the .o file will be put into the subdirectory named after the source file. For instance, with subdir-objects enabled, sub/dir/file.c will be compiled to sub/dir/file.o. Some projects prefer or require this mode of operation. You can specify subdir-objects in AUTOMAKE_OPTIONS (see [Changing Automake’s Behavior](https://www.gnu.org/software/automake/manual/html_node/Options.html)).

> When subdir-objects is specified, and source files which lie outside the current directory tree are nevertheless specified, as in foo_SOURCES = ../lib/other.c, Automake will still remove ../lib/other.o, in fact, ../lib/*.o **(e.g., at make clean, even though it is arguably wrong for one subdirectory to clean in a sibling. This may or may not be changed in the future.**

The problem is that we end up with duplicated clean targets for this single file if you specify it in multiple `makefile.am`s. I don't know how to fix this. @bwbarrett might.

Reply:
Yeah, we're mixing paradigms right now.  There's two paths out of this problem:

1. Don't set subdir-objects
2. Fold src/tuner/Makefile.am into src/Makefile.am (note, the source files don't have to move, just the contents of the Makefile).

I'd do (2), because it increases the amount of available parallelism in the makefile, because Make can build both the plugin and tuner files from the same make invocation.  Doing (1) means that we would have to have a makefile per source directory, which is not a big deal right now, but does decrease potential parallelism if we eventually clean up some of our makefiles.

Reply:
Ack. I folded the Makefile.am from the src/tuner into src/ . This seems to fix the build issues.